src/launcher/actions/appsActions.js

```diff
@@ -381,10 +385,11 @@ export function upgradeOfficialApp(name, version, source) {
 }
 
 export function launch(app) {
-    return () => {
+    return dispatch => {
         // The apps in state are Immutable Maps which cannot be sent over IPC.
         // Converting to plain JS object before sending to the main process.
         const appObj = app.toJS();
+        dispatch(sendLauncherUserData(`${EventAction.LAUNCH_APP} ${appObj.name}`, JSON.stringify(appObj)));
```

Comment:
Don't send the complete _appObj_, it may contain paths or username too. Just pick the relevant props like version, e.g..